### PR TITLE
libinput: 1.26.2 -> 1.27.0

### PR DESCRIPTION
**Pull request by trofi**
Changes: https://lists.freedesktop.org/archives/wayland-devel/2024-November/043860.html

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by jtojnar — APPROVED** [review]
Thanks.

- Changelog looks okay.
- Skimmed through [upstream diff](https://gitlab.freedesktop.org/libinput/libinput/-/compare/1.26.2...1.27.0), also fine.
- Weird that the [rebuild count](https://github.com/NixOS/nixpkgs/pull/323012) is so low compared to previously, possibly something broken in CI.

File: pkgs/by-name/sw/sway-unwrapped/package.nix [hunk 1]
```diff
@@ -31,6 +31,18 @@ stdenv.mkDerivation (finalAttrs: {
       inherit swaybg;
     })
 
+    # libinput-1.27 support:
+    #   https://github.com/swaywm/sway/pull/8470
+    (fetchpatch {
+      name = "libinput-1.27-p1.patch";
+      url = "https://github.com/swaywm/sway/commit/c8ea30122a0346ebf9959db194a78c8f677b2e0d.patch";
```
Comment (getchoo) [inline pkgs/by-name/sw/sway-unwrapped/package.nix:38]:
```suggestion
      url = "https://github.com/swaywm/sway/commit/bbadf9b8b10d171a6d5196da7716ea50ee7a6062.patch";
```
This have been merged and have in-tree commits now. We should probably use those instead

File: pkgs/by-name/sw/sway-unwrapped/package.nix [hunk 2]
```diff
@@ -31,6 +31,18 @@ stdenv.mkDerivation (finalAttrs: {
...
+      hash = "sha256-lA+oL1vqGQOm7K+AthzHYBzmOALrDgxzX/5Dx7naq84=";
+    })
+    (fetchpatch {
+      name = "libinput-1.27-p2.patch";
+      url = "https://github.com/swaywm/sway/commit/89723602d684f07885f29ce2606418f331185f38.patch";
```
Comment (getchoo) [inline pkgs/by-name/sw/sway-unwrapped/package.nix:43]:
```suggestion
      url = "https://github.com/swaywm/sway/commit/e2409aa49611bee1e1b99033461bfab0a7550c48.patch";
```

**Review by getchoo — CHANGES_REQUESTED** [review]

**Review by getchoo — APPROVED** [review]
Diff LGTM

**Pull request merged**